tools/walletextension/frontend/src/services/useGatewayService.ts

```diff
@@ -67,15 +79,16 @@ const useGatewayService = () => {
         showToast(ToastType.SUCCESS, "Added TEN Testnet");
       }
 
-      if (!(await isMetamaskConnected())) {
-        showToast(ToastType.INFO, "No accounts found, connecting...");
-        await connectAccounts();
-        showToast(ToastType.SUCCESS, "Connected to TEN Testnet");
-      }
-      await fetchUserAccounts();
+      await finaliseConnectAccounts()
+
     } catch (error: Error | any) {
-      showToast(ToastType.DESTRUCTIVE, `${error?.message}`);
-      throw error;
+      if (error.message === 'l is not a function') {
```

Comment:
Maybe worth a comment on why this hack was necessary if there's any context we could give to it.

Reply:
Ah also if finaliseConnectAccounts() threw an error  at this point would it fail silently because we don't catch it? Maybe worth a catch and log.